# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpConnectionUDP.cpp
# Commit: 8e7c97a37073
# Full Hash: 8e7c97a37073a14ba832e2f5c27f2c11b7787ef5
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 16:26:49
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r 5abfd5ad2e35 -r 8e7c97a37073 netwerk/protocol/http/HttpConnectionUDP.cpp
--- a/netwerk/protocol/http/HttpConnectionUDP.cpp	Wed Mar 15 12:41:26 2023 +0000
+++ b/netwerk/protocol/http/HttpConnectionUDP.cpp	Wed Mar 15 13:00:46 2023 +0000
@@ -77,6 +77,8 @@
     return NS_ERROR_FAILURE;
   }
   dnsAddrRecord->IsTRR(&mResolvedByTRR);
+  dnsAddrRecord->GetEffectiveTRRMode(&mEffectiveTRRMode);
+  dnsAddrRecord->GetTrrSkipReason(&mTRRSkipReason);
   NetAddr peerAddr;
   nsresult rv = dnsAddrRecord->GetNextAddr(mConnInfo->GetRoutedHost().IsEmpty()
                                                ? mConnInfo->OriginPort()
@@ -675,5 +677,11 @@
 
 bool HttpConnectionUDP::ResolvedByTRR() { return mResolvedByTRR; }
 
+nsIRequest::TRRMode HttpConnectionUDP::EffectiveTRRMode() {
+  return mEffectiveTRRMode;
+}
+
+TRRSkippedReason HttpConnectionUDP::TRRSkipReason() { return mTRRSkipReason; }
+
 }  // namespace net
 }  // namespace mozilla